eZ/Publish/Core/FieldType/Date/Value.php

```diff
@@ -39,10 +39,6 @@ class Value extends BaseValue
      */
     public function __construct(DateTime $dateTime = null)
     {
-        if ($dateTime !== null) {
-            $dateTime = clone $dateTime;
-            $dateTime->setTime(0, 0, 0);
-        }
```

Comment:
I have bad feelings about this change. For `ezdate` time part shouldn't be considered during the `\eZ\Publish\Core\FieldType\Date\Value` objects comparison which basically will change after the merge:

```php
$a = new Value(new DateTime());
sleep(1);
$b = new Value(new DateTime());

dump($a == $b) // currently: true, after the patch: false
```

Time truncating is also the main difference between `ezdatetime` and `ezdate` fields. 

So please double check behavior of 

* `\eZ\Publish\API\Repository\Values\Content\Query\Criterion\Field`
* `\eZ\Publish\API\Repository\Values\Content\Query\SortClause\Field`

with and without the patch using different APIs: PAPI, REST, GraphQL.